[receiver/prometheus] Add complete end-to-end configuration example

geekysatbir opened the pull request:
#### Description

Added a comprehensive end-to-end configuration example demonstrating how to use the Prometheus receiver in a production pipeline with processors (batch, resource) and exporters (otlp, prometheusremotewrite). This example helps users understand how to integrate the receiver into a complete observability setup.

The example includes:
- Prometheus receiver configuration with metric filtering using `metric_relabel_configs`
- Resource processor to add semantic convention attributes (`service.name`, `deployment.environment`)
- Batch processor for efficient metric batching
- Multiple export destinations (OTLP and Prometheus remote write)
- Complete service pipeline configuration

This addresses a common gap in documentation where component-specific examples don't show how to integrate components into a full pipeline.

#### Link to tracking issue

N/A - Documentation enhancement

#### Testing

- Verified the YAML syntax is valid
- Confirmed all component names and configurations follow OpenTelemetry Collector conventions
- Validated processor ordering (resource before batch) follows best practices
- Ensured endpoint formats are correct (gRPC for OTLP, HTTPS for Prometheus remote write)
- Cross-referenced with existing examples in other receiver READMEs for consistency

#### Documentation

Added a new "Complete Configuration Example" section to the Prometheus receiver README (lines 107-163) that demonstrates:

1. **Receiver Configuration**: Scrapes metrics from a service with filtering
2. **Processor Configuration**: 
   - Resource processor for adding semantic attributes
   - Batch processor for efficient metric export
3. **Exporter Configuration**:
   - OTLP exporter with gRPC endpoint
   - Prometheus remote write exporter with HTTPS endpoint
4. **Service Pipeline**: Complete metrics pipeline setup

The example follows OpenTelemetry best practices and provides a production-ready template that users can adapt for their own deployments.

geekysatbir committed:
[receiver/prometheus] Add complete end-to-end configuration example

Added a comprehensive configuration example demonstrating how to use the
Prometheus receiver in a production pipeline with processors (batch, resource)
and exporters (otlp, prometheusremotewrite). This example helps users understand
how to integrate the receiver into a complete observability setup.

dashpole commented on receiver/prometheusreceiver/README.md:
```diff
@@ -104,6 +104,63 @@ receivers:
+
+processors:
+  batch:
+    timeout: 10s
```
note that batching doesn't have an effect here.  There is a single scrape job with a scrape interval greater than the batch timeout.

geekysatbir replied:
Nice catch!! Made the changes.

dashpole commented on receiver/prometheusreceiver/README.md:
```diff
@@ -104,6 +104,63 @@ receivers:
+          scrape_interval: 15s
+          static_configs:
+            - targets: ['localhost:9090']
+          metric_relabel_configs:
```
maybe add a yaml comment describing what this is doing (filtering the metrics)

dashpole commented on receiver/prometheusreceiver/README.md:
```diff
@@ -104,6 +104,63 @@ receivers:
+      insecure: true
+  prometheusremotewrite:
+    endpoint: https://prometheus:9090/api/v1/write
+    external_labels:
```
we should probably just use resource attributes here instead of external_labels?

I also would rather keep the config for this exporter as simple as possible, given it may change in the future (and probably won't be updated here)

geekysatbir committed:
Address review comments: fix batch timing, add comments, remove redundant config

- Changed scrape_interval from 15s to 5s so batch processor is effective
- Added YAML comment explaining metric_relabel_configs filtering
- Removed redundant service.name from resource processor (already set by receiver)
- Removed external_labels from prometheusremotewrite to keep config simple
- Updated description to reflect changes
